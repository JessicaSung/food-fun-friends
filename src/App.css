.container {
  /* display: flex; */
  /* max-width: 60rem; */
  /* max-width: 1100px; */
  /* margin: 0 auto; */
  padding: 20px;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
}

.add-item {
  align-self: flex-start;
  background-color: #303050;
  border-radius: 2px;
  border-top: 3px solid #103D5D;
  box-shadow: 0 3px 4px 0 rgba(0, 0, 0, .14), 0 3px 3px -2px rgba(0, 0, 0, .12), 0 1px 8px 0 rgba(0, 0, 0, .2);
  max-width: 15rem;
  min-width: 15rem;
  /* margin: 0 auto; */
  padding: 60px 18px;
  width: 100%;
  /* margin-bottom: 34px; */
  /* margin-right: 50px; */
  /* margin-right: auto; */
  /* width: 25%; */
}

.display-item {
  /* display: flex; */
  width: 65%;
  margin-left: 50px;
}

.display-item button {
  margin-top: 20px;
  margin-left: auto;
  display: block;
}

.add-item form {
  width: 100%;
  overflow: hidden;
}

ul {
  display: flex;
  flex-wrap: wrap;
  margin: 0;
  padding: 0;
}

li {
  /* width: calc(50% - 50px); */
  width: 10rem;
  list-style-type: none;
  background-color: rgb(237, 237, 237);
  color: #000;
  margin-right: 50px;
  margin-bottom: 30px;
  box-shadow: 0 3px 4px 0 rgba(0, 0, 0, .14), 0 3px 3px -2px rgba(0, 0, 0, .12), 0 1px 8px 0 rgba(0, 0, 0, .2);
}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

body {
  font-family: 'Open Sans', sans-serif;
  background-color: lightblue;
}

header {
  color: #fff;
  background-color: #103D5D;
  padding: 33px 0 25px 30px;
}

header p {
  width: 400px;
  margin: 0 auto;
}

header .wrapper {
  display: flex;
  justify-content: space-between;
}

header h1 {
  font-size: 25px;
  font-weight: 600;
  letter-spacing: 0.3px;
  display: inline-block;
  margin: 0;
  vertical-align: middle;
}

.fa {
  font-size: 22px;
  transform: translateY(50deg);
}

header i {
  vertical-align: middle;
  margin-left: 12px;
}

.wrapper {
  max-width: 1100px;
  margin: 0 auto;
}

input {
  font-size: 22px;
  color: #000;
  padding: 18px 22px;
  font-size: 16px;
  margin-bottom: 17px;
  border: 0;
  display: block;
  width: 100%;
}

h2 {
  padding: 0;
  margin: 0;
  font-weight: 400;
}

li h3 {
  background-color: #103D5D;
  margin: 0;
  color: #fff;
  font-weight: 300;
  padding: 15px;
}

li p {
  padding: 15px;
}

form button {
  width: 100%;
  margin-top: 10px;
}

button {
  border-radius: 2px;
  min-width: 88px;
  background-color: #FA6900;
  cursor: pointer;
  border: 0;
  min-width: 120px;
  color: #fff;
  font-size: 14px;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, .26);
  padding: 10px 6px;
}

.side {
  flex: 1;
}

.user-profile {
  position: fixed;
  bottom: 20px;
  right: 40px;
  width: 100px;
  height: 100px;
}

.user-profile img {
  border-radius: 50%;
  max-width: 100%;
}

.authentication-button {
  float: right;
  margin-right: 50px;
}