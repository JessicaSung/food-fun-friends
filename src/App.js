import React, { Component } from 'react';
import './App.css';
import firebase, { auth, provider } from './firebase.js';

class App extends Component {
  state = {
    currentItem: '',
    username: '',
    items: [],
    user: null
  }
  handleChange = (e) => {
    // console.log(e, e.target.name, e.target.value);

    // The e.persist(); is necessary with an updater function for setState because otherwise SyntheticEvent is pooled.
    e.persist();
    this.setState(() => ({ [e.target.name]: e.target.value }));
    // this.setState({ [e.target.name]: e.target.value });
    // console.log(this.state);
  }
  handleSubmit = (e) => {
    e.preventDefault();
    const itemsRef = firebase.database().ref('items');
    const item = {
      title: this.state.currentItem,
      user: this.state.username
    }
    itemsRef.push(item);
    this.setState(() => ({
      currentItem: '',
      username: ''
    }));
  }
  removeItem = (itemId) => {
    const itemRef = firebase.database().ref(`/items/${itemId}`);
    itemRef.remove();
  }
  componentDidMount() {
    const itemsRef = firebase.database().ref('items');
    itemsRef.on('value', (snapshot) => {
      let items = snapshot.val();
      let newState = [];
      for (let item in items) {
        newState.push({
          id: item,
          title: items[item].title,
          user: items[item].user
        });
      }
      this.setState(() => ({
        items: newState
      }));
    });
  }
  render() {
    return (
      <div className='app'>
        <header>
          <h1>Food Fun Friends</h1>
        </header>
        <div className="container">
          <section className='add-item side'>
            <form onSubmit={this.handleSubmit}>
              <input type="text" name="username" placeholder="Your name" onChange={this.handleChange} value={this.state.username} />
              <input type="text" name="currentItem" placeholder="What are you bringing?" onChange={this.handleChange} value={this.state.currentItem} />
              <button>Add Item</button>
            </form>
          </section>
          <section className='display-item side'>
            <ul>
              {this.state.items.map((item) =>
                <li key={item.id}>
                  <h3>{item.title}</h3>
                  <p>brought by: {item.user}</p>
                  <button onClick={() => this.removeItem(item.id)}>Remove Item</button>
                  {/* This triggers error message: Cannot update during an existing state transition (such as within `render` or another component's constructor). */}
                  {/* <button onClick={this.removeItem(item.id)}>Remove Item</button> */}
                </li>
              )}
            </ul>
          </section>
        </div>
      </div>
    );
  }
}

export default App;
