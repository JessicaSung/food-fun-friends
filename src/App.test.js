import React from 'react';
import ReactDOM from 'react-dom';
import { toMatchImageSnapshot } from 'jest-image-snapshot';
import puppeteer from 'puppeteer';
import renderer from 'react-test-renderer';

import App from './App';

// Create React App default test.
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// jest-image-snapshot + puppeteer saves a png snapshot and compares.
expect.extend({ toMatchImageSnapshot });

describe('jest-image-snapshot and puppeteer png match', () => {
  let browser;

  beforeAll(async () => {
    browser = await puppeteer.launch();
  });

  it('should match the visual layout', async () => {
    const page = await browser.newPage();
    await page.goto('http://localhost:3000/');
    // await page.setViewport({
    //   width: 800,
    //   height: 1200
    // });

    const image = await page.screenshot({
      // fullPage: true
    });

    expect(image).toMatchImageSnapshot();
  });

  afterAll(async () => {
    await browser.close();
  });
});

// react-test-renderer saves an html structure snapshot and compares.
test('react-test-renderer html match', () => {
  const component = renderer.create(
    <App />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
