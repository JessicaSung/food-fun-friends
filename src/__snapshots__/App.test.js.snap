// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-test-renderer html match 1`] = `
<div
  className="app"
>
  <header>
    <h1>
      Food Fun Friends
    </h1>
  </header>
  <div
    className="container"
  >
    <section
      className="add-item side"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          name="username"
          onChange={[Function]}
          placeholder="Your name"
          type="text"
          value=""
        />
        <input
          name="currentItem"
          onChange={[Function]}
          placeholder="What are you bringing?"
          type="text"
          value=""
        />
        <button>
          Add Item
        </button>
      </form>
    </section>
    <section
      className="display-item side"
    >
      <ul>
        <li>
          <h3>
            Zongzi
          </h3>
          <p>
            brought by: 
            Aileen
          </p>
          <button
            onClick={[Function]}
          >
            Remove Item
          </button>
        </li>
        <li>
          <h3>
            Pad Thai
          </h3>
          <p>
            brought by: 
            Brian
          </p>
          <button
            onClick={[Function]}
          >
            Remove Item
          </button>
        </li>
        <li>
          <h3>
            Curry Chicken
          </h3>
          <p>
            brought by: 
            Jessica
          </p>
          <button
            onClick={[Function]}
          >
            Remove Item
          </button>
        </li>
      </ul>
    </section>
  </div>
</div>
`;
